lib/modules/RESTips.js

```diff
@@ -211,142 +281,106 @@ const tips = [{
-	$('body').on('click', '#disableDailyTipsCheckbox', (e: Event) => Options.set(module, 'dailyTip', (e.target: any).checked));
+const dailyTipsCheckbox = _.once(() =>
+	$(`<label> <input type="checkbox" name="disableDailyTipsCheckbox" ${module.options.dailyTip.value ? 'checked' : ''} />Show these tips once every 24 hours</label>`)
+		.click((e: Event) => Options.set(module, 'dailyTip', (e.target: any).checked))
+);
 
-	// create the special "you have never visited the console" guider...
-	createGuider(0, 'console');
-	for (const i of range(0, tips.length)) {
-		createGuider(i);
-	}
-});
+let currTipIndex = 0;
 
-function createGuider(i, special) {
-	let thisID, thisTip;
+function showOrdinaryTip(change?: 'random' | 'prev' | 'next') {
+	let tip;
 
-	if (special === 'console') {
-		thisID = special;
-		thisTip = consoleTip;
-	} else {
-		thisID = `tip${i}`;
-		thisTip = tips[i];
+	PenaltyBox.alterFeaturePenalty(module.moduleID, 'dailyTip', -20);
+
+	while (!tip || (tip.attachTo && !$(tip.attachTo).is(':visible'))) {
+		if (change === 'random') currTipIndex = _.random(tips.length);
+		else if (change === 'prev') currTipIndex -= 1;
+		else if (change === 'next') currTipIndex += 1;
+		else change = 'next'; // For next iteration
+		currTipIndex = positiveModulo(currTipIndex, tips.length);
+		tip = tips[currTipIndex];
 	}
-	const title = generateTitle(thisTip);
-	const len = tips.length;
-	const $description = generateContent(thisTip);
-	const nextidx = ((parseInt(i + 1, 10)) >= len) ? 0 : (parseInt(i + 1, 10));
-	const nextID = `tip${nextidx}`;
-	const thisChecked = (module.options.dailyTip.value) ? 'checked="checked"' : '';
-
-	const guiderObj = {
-		attachTo: thisTip.attachTo || undefined,
+
+	const $description = generateContent(tip);
+	$description.on('click', 'a', () => { PenaltyBox.alterFeaturePenalty(module.moduleID, 'sectionMenu', -20); });
+
+	showTip({
 		buttons: [{
 			name: 'Prev',
-			onclick: prevTip,
+			onclick: () => showOrdinaryTip('prev'),
 		}, {
 			name: 'Next',
-			onclick: nextTip,
+			onclick: () => showOrdinaryTip('next'),
 		}],
 		onClose: () => { PenaltyBox.alterFeaturePenalty(module.moduleID, 'sectionMenu', 15); },
 		description: $description,
-		buttonCustomHTML: `<label class="stopper"> <input type="checkbox" name="disableDailyTipsCheckbox" id="disableDailyTipsCheckbox" ${thisChecked} />Show these tips once every 24 hours</label>`,
-		id: thisID,
-		next: nextID,
-		position: tips[i].position || undefined,
-		xButton: true,
-		title,
-	};
-	if (special === 'console') {
-		delete guiderObj.buttonCustomHTML;
-		delete guiderObj.next;
-		delete guiderObj.buttons;
-
-		guiderObj.title = 'RES is extremely configurable';
-	}
-
-	$description.on('click', 'a', () => { PenaltyBox.alterFeaturePenalty(module.moduleID, 'sectionMenu', -20); });
-
-	guiders.createGuider(guiderObj);
-}
-
-function handleEscapeKey(event: KeyboardEvent) {
-	if (event.which === 27) {
-		PenaltyBox.alterFeaturePenalty(module.moduleID, 'sectionMenu', 10);
```

Comment:
Yeah, it looks like it was appropriate to trash this since `onClose` will handle it, and `sectionMenu` should be replaced with `dailyTip` I guess..

... or maybe we should strip out the penaltybox handling in this module.